lib/info-extras.js

```diff
@@ -124,11 +124,35 @@ exports.getPublished = (body) => {
  * @return {Array.<Object>}
  */
 exports.getRelatedVideos = (body) => {
-  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': {"rvs":', '},');
+  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': ', '\n');
+  let watchNextJson;
   try {
+    jsonStr = jsonStr.substring(0, jsonStr.length - 1);
     jsonStr = JSON.parse(jsonStr);
-  } catch (err) {
+    watchNextJson = JSON.parse(jsonStr.watch_next_response);
+  }
+  catch (err) {
     return [];
   }
-  return jsonStr.split(',').map((link) => qs.parse(link));
-};
+  let secondaryResults = watchNextJson.contents.twoColumnWatchNextResults.secondaryResults.secondaryResults.results;
+  let recomendedVideos = [];
+  for (let result of secondaryResults) {
+    if (result.compactVideoRenderer) {
+      let details = result.compactVideoRenderer;
+      try {
+        recomendedVideos.push({
+          id: details.videoId,
+          title: details.title.simpleText,
+          author: details.shortBylineText.runs[0].text,
+          ucid: details.shortBylineText.runs[0].navigationEndpoint.browseEndpoint.browseId,
+          author_thumbnail: details.channelThumbnail.thumbnails[0].url,
+          short_view_count_text: details.shortViewCountText.simpleText,
+          view_count: details.viewCountText.simpleText.replace(' views', ''),
```

Comment:
`shortViewCountText` can be any of:
```javascript
{"shortViewCountText": {"simpleText": "2.5M views"}}
{"shortViewCountText": {"simpleText": "Recommended for you"}} // Doesn't appear to be handled
{"shortViewCountText": {"runs": [{"text": "47"}, {"text": " watching"}]} // Doesn't appear to be handled
```

For "Recommended for you", it's still possible to provide a string in `X views` by retrieving the corresponding object from `rvs` as part of the `jsonStr` in `getRelatedVideos`.

Reply:
Thank you for the review. Can you please check the latest commit